### PR TITLE
ff - outro

### PR DESCRIPTION
**Pull request by woozyking**
venn

profitable

branching

### CODE REVIEW

File: ff-outro.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+## git
+
+Among many tools that support the open-source software ecosystem, `git` is the most popular one that provides efficient ways to distribute and manage source code. Its decentralized nature allows every contributor to fully access the working copy and the entire change history and derivatives.
+
+The collaboration happens in a peer-to-peer fashion. Collaborators branch out the source code to implement new features or issue fixes. Upon concluding the implementation, the contributor requests to merge back to the mainstream working copy of the source code, attract others to review, and offer suggestions to improve. When enough contributors reach a consensus, the implementation gets merged back to the mainstream. During this entire process, automation plays a critical role in simplifying every participant's experience through automated testing and suggestions, so contributors do not waste time on trivial problems such as code-style differences and logical flaws. After the merge, it is common to see automated deployment or release of the software to a designated audience.
```
Comment (tlowande) [inline ff-outro.md:41]: is `branch out` a jargon that we should further clarify for non-devs? Maybe `fork` and `merge` could be illustrated with those [diagrams](https://www.bogotobogo.com/cplusplus/images/Git/Fast_Forward_Merge/TypicalMerge.png ) images - from https://www.bogotobogo.com/DevOps/SCM/Git/Git_GitHub_Fast-Forward_Merge.php?

File: ff-outro.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+This redefinition gives developers better perspectives and a higher sense of ownership, allowing them to think and act more strategically, which improves the collective pool of wisdom for balanced decision-making.
+
+### Acceleration through Machine Learning
+
+Open-source projects such as [TensorFlow](https://www.tensorflow.org/) and [PyTorch](https://pytorch.org/) offer a set of sophisticated implementation of modern machine-learning algorithms and utilities that accelerate the path from AI model research to production.
+
+There are also AI models that accelerate the elimination of many rudimentary job functions. Such as [comprehending and summarizing lengthy text](https://openai.com/blog/better-language-models/),  [elaborating from seeding phrases](https://app.inferkit.com/demo), or [categorize software changes]((https://medium.com/locus-engineering/road-to-automation-release-notes-d1c49cc97d9)). These technologies can procedurally generate video game stages and contents to offer a unique and challenging experience or learn to [beat]((https://deepmind.com/research/case-studies/alphago-the-story-so-far)) real [players](https://openai.com/projects/five/).
```
Comment (tlowande) [inline ff-outro.md:89]: Is ML a topic you intend to develop in the future or are you using it to exemplify opensource and abstractions? Although well explained, I missed the point if not only exemplify - I feel like you have a goal with this 🤔 ...

Comment (tlowande) [inline ff-outro.md:89]: Is ML a topic you intend to develop in the future or are you using it to exemplify opensource and abstractions? Although well explained, I missed the point if not only exemplify - I feel like you have a goal with this 🤔 ...

**Review by tlowande — COMMENTED** [review]
open source is a great topic to bring up async develop and git 👍 
Very well contextualized.
> next in the series should be called `mais um` (another one)

File: ff-outro.md (hunk 1 above)
Comment (woozyking) [reply ff-outro.md:41]: You’re right, image illustrations would help a lot! I’ll add those later

File: ff-outro.md (hunk 2 above)
Comment (woozyking) [reply ff-outro.md:89]: You’re right on both. This should be associated closer to open-source/abstraction, I’ll work on the phrasing more around that point. I should also follow up with another minor section on what will be brought up in future sessions which will indeed include ML stuff, which would feel more natural as a closure of this entire outro

**Comment by woozyking** [comment]
> open source is a great topic to bring up async develop and git 👍
> Very well contextualized.
> 
> > next in the series should be called `mais um` (another one)

Yes, you got it! Thank you for reviewing!

**Review by tlowande — APPROVED** [review]
Beaaauty 🚀

File: ff-outro.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+Such an interpretation of open-source practices allows the company to attract top talents worldwide to collaborate on a service platform enjoyed by their users. By 2010 (a little over three years since inception), GitHub has claimed to be the largest host of source code in the world until today.
+
+## EQ Works - our adaptation of open-source
+
+A key idea behind the open-source movement is to leverage their learnings to make information and curation more transparent and efficient. This social movement coincides with the phenomenon of the internet-driven information explosion and the increasing realization of how closed platforms could [unethically profit from manipulating the information flow and shape](https://en.wikipedia.org/wiki/Facebook%E2%80%93Cambridge_Analytica_data_scandal).
```
Comment (maluhoss) [inline ff-outro.md:85]: suggestion for "This social movement coincides with the phenomenon of the internet-driven information explosion and the increasing realization of how closed platforms could [unethically profit from manipulating the information flow and shape]": 

"This social movement coincides with increased access to information online and growing realization of how closed platforms could unethically profit from manipulation the information flow and shape."

File: ff-outro.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+As a part of the workspace meta shift, more organizations start to realize the benefits of open-source software and their community practices. Since around the beginning of 2014, our dev team has started to build products using open-source technologies and evolve our culture by adapting the open-source ways.
+
+### Round-table wisdom
+
+Instead of on-paper seniority, materialized merits become the only valid argument between project contributors in discussions and debates. We demonstrate cases in a verifiable way, rapidly iterate based on feedback, and make frequent releases with peer-review and automation that capture trivial issues.
+
+The practice allows us to eliminate hierarchy and processes to boost productivity and innovation opportunities while maintaining a flat and fluid team. We minimize constraints to give everyone the freedom to pursue after their interest and the ability to develop, test and deploy their implementations without waiting for others.
```
Comment (maluhoss) [inline ff-outro.md:93]: Suggestion for the first line: "The practice allows us to eliminate hierarchy and enable processes that boost productivity and innovation through a flat and fluid team."

File: ff-outro.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+### Respect Time - Work Alone, Together
+
+The transparent and asynchronous model of information distribution allows our collaboration and communication to be in a manner that respects each other's time. With the information availed to us, each team member infers and summarizes on their own time to form most decisions without bothering or getting blocked by others.
```
Comment (maluhoss) [inline ff-outro.md:97]: think you meant 'With the information available to us..." for the second line

**Review by maluhoss — COMMENTED** [review]
most suggestions are around rewording to make some sentences a bit more digestible - sometimes there's a lot of words that can take a little longer to conceptualize. Tried my best to keep the initial thought. Hope this helps!

**Comment by maluhoss** [comment]
Overall, great work! Might help to have an ongoing example to jump back to when explaining the different aspects of open-source technology like jupyter notebook. but something that is readily relatable to the audience. I used google docs a few times, so you could use an example like google docs and highlight how it could be augmented if it were open-source

File: ff-outro.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+![sdlc](https://upload.wikimedia.org/wikipedia/commons/thumb/1/19/SDLC_-_Software_Development_Life_Cycle.jpg/596px-SDLC_-_Software_Development_Life_Cycle.jpg)
+
+(Software Development Life Cycle from [Wikimedia Commons](https://commons.wikimedia.org/wiki/File:SDLC_-_Software_Development_Life_Cycle.jpg))
+
+Empowered by transparency, the open-source community often devises pragmatic and objective-focused methodologies, eliminating as many intermediaries and manual processes as possible. Contributors can infer as much as possible from available information before involving others. These methodologies continuously evolve to adapt to the people who make and use the software instead of the conventional corporate world's rigid models. The transparency also generates more data that drives more opportunities to help make data-driven decisions.
```
[the remark this answers is not in the document]
Comment (woozyking) [reply ff-outro.md:31]: kind of need a word to signal the diff between that and the "manual processes" portion 🤔

File: ff-outro.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+### Asynchronous and Decentralized Workforce
+
+Open-source participants jointly form a workforce that resembles an efficient network system that is asynchronous and decentralized. With an abundance of information, minimal processes, and a growingly sophisticated set of tools and automation, open-source contributors can collaborate efficiently at any time and in any location.
```
[the remark this answers is not in the document]
Comment (woozyking) [reply ff-outro.md:35]: they don't form an external workforce, but a new type of collective workforce though. i can probably go with "Open-source contributors form a workforce that is asynchronous and decentralized", agree it'd look simpler and no need for an analogy of "network system".

File: ff-outro.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+To sustain an asynchronous and decentralized workforce with a massive amount of information exposed through transparency, open-source participants frequently make and improve automation to optimize or abstract away processes. This mindset drives people to continually lookout for new ways to adapt and improve their workflow.
+
+But it is rare for existing tools and automation to provide a perfect fit for everyone's use-cases. People resort to hacking, or creative workarounds, to bend tools to their will.
+
+The hacking culture also provides an organic counter-balance to some problems that plague many software projects, such as under-documentation, lack of testing, and [leaky abstractions](https://en.wikipedia.org/wiki/Leaky_abstraction).
+
+## git
+
+Among many tools that support the open-source software ecosystem, `git` is the most popular one that provides efficient ways to distribute and manage source code. Its decentralized nature allows every contributor to fully access the working copy and the entire change history and derivatives.
```
[the remark this answers is not in the document]
Comment (woozyking) [reply ff-outro.md:59]: derivatives are branches, tags and all other git refs.

i would not compare google docs with git. google docs is version controlled, but not distributed/decentralized.

File: ff-outro.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (maluhoss) [reply ff-outro.md:31]: I see what you mean. maybe say 'practical and objective-focused approach'

File: ff-outro.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (maluhoss) [reply ff-outro.md:59]: oh I see. maybe say 'access the current copy that's live to the public, the history of changes made to the current one and other working copies that have been created by others. basically some other word than derivatives.

Also i was referring the history of changes to google doc keeping track of past changes, not product to product (google docs to git).

File: ff-outro.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+The collaboration happens in a peer-to-peer fashion. Collaborators branch out the source code to implement new features or issue fixes. Upon concluding the implementation, the contributor requests to merge back to the mainstream working copy of the source code, attract others to review, and offer suggestions to improve. When enough contributors reach a consensus, the implementation gets merged back to the mainstream. During this process, automation allows contributors to not waste time on trivial problems such as code-style differences and logical flaws. After the merge, it is common to see automated deployment or release of the software to a designated audience.
+
+![branching](https://user-images.githubusercontent.com/2837532/108411648-e0107300-71f6-11eb-8148-d66dbd4d19d5.png)
+
+(From [Understanding the GitHub flow](https://guides.github.com/introduction/flow/))
+
+When unresolvable disagreements break out, one can maintain the branch derivative as a permanent _fork_ to evolve the software project to a different trajectory. The practice of forking is why we have a diverse number of Linux distributions.
```
[the remark this answers is not in the document]
Comment (woozyking) [reply ff-outro.md:67]: 🤔

File: ff-outro.md (hunk 4 above)
Comment (woozyking) [reply ff-outro.md:93]: elimination of processes is intended. will rephrase as

"The practice allows us to eliminate hierarchy and processes to enable productivity and innovation through a flat and fluid team."

File: ff-outro.md (hunk 5 above)
Comment (woozyking) [reply ff-outro.md:97]: "[avail](https://en.wiktionary.org/wiki/avail)" is intended

**Comment by woozyking** [comment]
> Overall, great work! Might help to have an ongoing example to jump back to when explaining the different aspects of open-source technology- like jupyter notebook. but something that is readily relatable to the audience. I used google docs a few times, so you could use an example like google docs and highlight how it could be augmented if it were open-source

good point on examples (though comparing google docs with git is not the best). we can keep this ongoing to add more in the future. though as a high-level outro->intro, the materialized examples are to come in the next series.

File: ff-outro.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (woozyking) [reply ff-outro.md]: ye the version control aspects for sure

**Pull request merged**